src/web3mail/IExecWeb3mail.ts

```diff
@@ -37,11 +37,14 @@ export class IExecWeb3mail {
   private graphQLClient: GraphQLClient;
 
   constructor(
-    ethProvider: Eip1193Provider | Web3SignerProvider,
+    ethProvider?: Eip1193Provider | EnhancedWallet | string,
```

Comment:
I would not expose `EnhancedWallet` which is supposed to be low level iexec SDK here
```suggestion
    ethProvider?: Eip1193Provider | Web3SignerProvider | string,
```